src/settings.cpp

```diff
@@ -36,6 +36,94 @@ settings::settings()
 {
 }
 
+settings::settings(config::settings context)
+  : settings()
+{
+    switch (context)
+    {
+        case config::settings::mainnet:
+        {
+            encoded_genesis_block =
```

Comment:
The settings pattern uses serializable classes to read settings to produce the corresponding object. I think the proper generalization here is actually a chain::block.

Reply:
This would make the `chain::block::genesis()` function obsolete. I'll remove it then.

Reply:
It makes it redundant, but that’s fine. It can accept only the standard config enumeration and construct the well know instances, as it does now, and those can be used to simplify the standard settings init.